### ISSUE TITLE
BUG: vpp-networking fails to attach ports with pre-assigned mac addresses to VMs 

### ISSUE THREAD

**Issue by taylor** [cncf/cnf-testbed#189]
Creating ports with specific MAC for ports that are assigned to VMs fails to work as expected.  The VM never sees the ports

- [x] Investigate and recreate
  * Disabling port security when creating the OpenStack network fixes issue
- [x] Update scripts to disable port security

**Comment by taylor** [comment]
@robertstarmer please document the issue and create a upstream vpp-networking ticket.

/cc @denverwilliams

**Comment by robertstarmer** [comment]
I'm not able to reproduce this issue.  In a recent deployment, we're able to add a VM with external ports and random MAC addresses.  We're using the following script to create the ports/attach and launch a VM with them attached.

``` bash
#!/bin/bash

if [ $# -lt 3 ]; then
  echo "ERROR: this script requires 3 parameters"
  echo "USAGE: $0 <server_name> <vlan_1_id> <vlan_2_id>"
  echo "NOTE: The script assumes networks are named vlan<vlan_id>"
  exit
fi


source ~/openrc
cat > /tmp/${1}.cfg <<EOF
#!/bin/bash
passwd ubuntu <<EOL
passwordthatshouldbechanged
passwordthatshouldbechanged
EOL
EOF

p1=$(openstack port create s${1}p1 --network vlan${2} --mac-address $(echo fa:17:b4:$[RANDOM%10]$[RANDOM%10]:$[RANDOM%10]$[RANDOM%10]:$[RANDOM%10]$[RANDOM%10]) | awk '/ id / {print $4}')
p2=$(openstack port create s${1}p2 --network vlan${3} --mac-address $(echo fa:17:b4:$[RANDOM%10]$[RANDOM%10]:$[RANDOM%10]$[RANDOM%10]:$[RANDOM%10]$[RANDOM%10]) | awk '/ id / {print $4}')
openstack server create ${1} --flavor c0.small --image xenial --nic port-id=${p1} --nic port-id=${p2} --config-drive True --user-data /tmp/${1}.cfg
```
@bvera @taylor @denverwilliams 

We are building a new cluster in order to validate the current test results.

**Comment by robertstarmer** [comment]
Added a version of the above script (create_dual_mac.sh) to the openstack_scripts role, tested with random, one, or both addresses being added.

**Comment by taylor** [comment]
@denverwilliams @michaelspedersen please verify this works for you

**Comment by michaelspedersen** [comment]
Based on a previous comment by @denverwilliams trying to assign a MAC and it failing causes quite extensive problems (server/cluster breaks), so I have not tested it in the environment I have been using for benchmarking preparations.

I can however add that we have a workaround that works without the ability to assign MAC addresses to ports ([create_instance_triple_port.sh](https://github.com/cncf/cnf-testbed/blob/master/comparison/kubecon18-chained_nf_test/VNF_openstack/create_instance_triple_port.sh)). From my current testing this works out well, but it does add a bit more work in configuring the traffic generator with the MACs of each chain endpoint.

**Comment by robertstarmer** [comment]
I believe this is addressed by an update to the latest etcd (v3.3.12), which code for which was committed previously. @taylor @denverwilliams @michaelspedersen

**Comment by taylor** [comment]
Disabling port security fixes issue.

Ian said this should work with port security  on.  Will open new ticket to investigate using port security when creating new networks.
